src/main/java/uk/gov/hmcts/ccd/data/casedetails/search/builder/AccessControlGrantTypeQueryBuilder.java

```diff
@@ -0,0 +1,77 @@
+
+@Component
+@SuppressWarnings("checkstyle:AbbreviationAsWordInName")
+public class AccessControlGrantTypeQueryBuilder {
+
+    private static final String QUERY = "( %s )";
+    private static final String FINAL_QUERY = " AND ( %s )";
+
+    private final BasicGrantTypeQueryBuilder basicGrantTypeQueryBuilder;
+    private final SpecificGrantTypeQueryBuilder specificGrantTypeQueryBuilder;
+    private final StandardGrantTypeQueryBuilder standardGrantTypeQueryBuilder;
+    private final ChallengedGrantTypeQueryBuilder challengedGrantTypeQueryBuilder;
+    private final ExcludedGrantTypeQueryBuilder excludedGrantTypeQueryBuilder;
+
+    public AccessControlGrantTypeQueryBuilder(BasicGrantTypeQueryBuilder basicGrantTypeQueryBuilder,
+                                              SpecificGrantTypeQueryBuilder specificGrantTypeQueryBuilder,
+                                              StandardGrantTypeQueryBuilder standardGrantTypeQueryBuilder,
+                                              ChallengedGrantTypeQueryBuilder challengedGrantTypeQueryBuilder,
+                                              ExcludedGrantTypeQueryBuilder excludedGrantTypeQueryBuilder) {
+        this.basicGrantTypeQueryBuilder = basicGrantTypeQueryBuilder;
+        this.specificGrantTypeQueryBuilder = specificGrantTypeQueryBuilder;
+        this.standardGrantTypeQueryBuilder = standardGrantTypeQueryBuilder;
+        this.challengedGrantTypeQueryBuilder = challengedGrantTypeQueryBuilder;
+        this.excludedGrantTypeQueryBuilder = excludedGrantTypeQueryBuilder;
+    }
+
+    public String createQuery(List<RoleAssignment> roleAssignments, Map<String, Object> params) {
+        String basicQuery = basicGrantTypeQueryBuilder.createQuery(roleAssignments, params);
+        String specificQuery = specificGrantTypeQueryBuilder.createQuery(roleAssignments, params);
+        String standardQuery = standardGrantTypeQueryBuilder.createQuery(roleAssignments, params);
+        String challengedQuery = challengedGrantTypeQueryBuilder.createQuery(roleAssignments, params);
+
+        String orgQuery = formatQuery(standardQuery, challengedQuery);
+
+        String tmpQuery = orgQuery;
+        String excludedQuery = excludedGrantTypeQueryBuilder.createQuery(roleAssignments, params);
+        if (StringUtils.isNotBlank(excludedQuery)) {
+            tmpQuery = tmpQuery
+                + getOperator(tmpQuery, " AND NOT ")
+                + excludedQuery;
+            tmpQuery = String.format(QUERY, tmpQuery);
+        }
+
+        String nonOrgQuery = formatQuery(basicQuery, specificQuery);
+        if (StringUtils.isNotBlank(nonOrgQuery)) {
+            tmpQuery = nonOrgQuery
+                + getOperator(tmpQuery, " OR ")
+                + tmpQuery;
+        }
+
+        return StringUtils.isNotBlank(tmpQuery) ? String.format(FINAL_QUERY, tmpQuery) : tmpQuery;
+    }
+
+    private String formatQuery(String queryOne,
+                               String queryTwo) {
```

Comment:
Minor if you think worth doing.. To try and make it a bit clearer what it actually does, and passing in the desired operator (normally `" OR "`) 
```suggestion
    private String mergeQuery(String queryOne,
                               String queryTwo,
                               String operator) {
```

Reply:
done